Src/IronPython/Runtime/PythonAsciiEncoding.cs

```diff
@@ -17,44 +18,62 @@ namespace IronPython.Runtime {
     /// </summary>
     [Serializable]
     sealed class PythonAsciiEncoding : Encoding {
+        // Singleton (global) instances are readonly, so their fallbacks cannot be accidentally modified unless cloned
         internal static readonly Encoding Instance = MakeNonThrowing();
         internal static readonly Encoding SourceEncoding = MakeSourceEncoding();
 
         internal PythonAsciiEncoding()
             : base() {
         }
 
-        internal static Encoding MakeNonThrowing() {
-            Encoding enc = new PythonAsciiEncoding();
+        internal PythonAsciiEncoding(EncoderFallback encoderFallback, DecoderFallback decoderFallback)
+#if NETCOREAPP2_1
```

Comment:
Should be either `#if NETCOREAPP2_1 || NETSTANDARD2_0` or `#if !NET45`. Second one might be better so we don't have to mess with it when we add new targets.

Reply:
Will do.

Reply:
Is `NET45` also defined for Mono? And if not, what is defined for Mono?

Reply:
Yes, `NET45` is also for Mono. Looking at this again, why do we need reflection here? Can't we just do:
```C#
EncoderFallback = encoderFallback;
DecoderFallback = decoderFallback;
```

Reply:
This won’t work because every newly instantiated `Encoding` is read-only and it should stay so if used as a global object.

Reply:
I see, that is quite unfortunate. Does Mono use the same private member names? Probably not much of an issue if we're going to (hopefully) drop `PythonAsciiEncoding`... Maybe we should just drop NET45 and make the minimum supported version NET461?

Reply:
Yes, Mono has the same private names and unfortunately the desired constructor internal. I have changed the condition to `!NET45`, this will do for now. This seems to me a too small reason to drop .NET 4.5 support.